xml/hwbp_registers.xml

```diff
@@ -801,6 +808,11 @@
         useful both for external debuggers and for software monitor programs.
         For that case it is not necessary to support \FcsrIcountCount greater
         than 1.
+
+        If \RcsrIcount is only implemented for single step, then it can be done
+        using a single bit that ties \FcsrIcountM, \FcsrIcountS, \FcsrIcountU,
+        \FcsrIcountVs, and \FcsrIcountVu together while \FcsrIcountCount is
+        hard-wired to 1.
```

Comment:
Plus a pending bit?

Consider this:
- mstatus.MIE=0 and we're in M mode
- write icount action=0, m=s=u=vs=vu=1 (i.e. set that single bit), count=1 (which is hard-wired), pending=0
- execute a bunch of instructions.
- read icount.  I would expect to see pending=1.
- write mstatus.MIE=1
- I would expect the trap to happen before the instruction following the MIE write because pending is already set.  If pending is hard-wired (under the assumption that it will always fire immediately) then I think it would trap one instruction later.

Perhaps "a mode where the trigger is enabled" (up above) needs to be clarified with respect to this MIE behavior.  I'm assuming that icount is updated regardless of MIE and that only the trigger is suppressed.  But I think others could interpret it differently.

Reply:
This section is already wrong. You can't tie all the mode bits together and have a single step useful to native debuggers. I'm redoing it.

Reply:
In fact, I'm mostly getting rid of it. The single-bit implementation can work, but only if you tie all mode bits together and have max count of 1. If all mode bits are enabled together, there is no case where the existence of a pending bit can make a different. But this is only useful for an external debugger, and we already say that for external debuggers dcsr.step is the preferred solution.

For native debugging, the least hardware I can think of is 2 bits, because there are effectively 3 states: disabled, pending, and enabled. This assumes the mode bits are tied together in such a way that there is only one sensible option. I'm not sure if that's even worth mentioning anywhere. It would be a long explanation with little benefit.